### ISSUE TITLE
Re-implement unit changing

### ISSUE THREAD

**Issue by nmearl** [nmearl/pyfocal#14]
Copy over code for changing the astropy quantity unit for displayed plots.

**Comment by pllim** [comment]
Regarding flux units, I don't think Astropy can convert properly between magnitudes and linear flux yet, not from STMag to ABMag. Is there a reason not to use pysynphot like JWST ETC? See http://ssb.stsci.edu/pysynphot/docs/units.html

**Comment by nmearl** [comment]
Interesting. I was under the impression that astropy quantities was able to do the conversion. Here's a chunk from [their docs](http://docs.astropy.org/en/stable/units/logarithmic_units.html):

```python
>>> B, V = b_i0 + zp_b, v_i0 + zp_v
>>> B, V  
(<Magnitude [ 17.2       , 20.89280314, 22.95257499] mag(ST)>,
 <Magnitude [ 17.        , 21.1195437 , 22.51029996] mag(ST)>)
>>> V.to(u.ABmag, u.spectral_density(5500.*u.AA))  
<Magnitude [ 16.99023831, 21.10978201, 22.50053827] mag(AB)>
```

Is this what you mean?

**Comment by pllim** [comment]
I meant converting from magnitude to, say, `f_lambda`, or vice versa.

**Comment by nmearl** [comment]
Ah, I see. Yes, looking at [pysynphot's units section], it seems they just make a bunch of custom astropy units. Perhaps it would be better to use pysynphot instead for unit conversions.

**Comment by pllim** [comment]
Don't get me wrong, I think Astropy units is very promising. And it is fine as long as you don't need to convert between magnitudes and linear flux units.

**Comment by ibusko** [comment]
"Is there a reason not to use pysynphot like JWST ETC? "

I understand that the JWST ETC does not use pysynphot at all.

**Comment by pllim** [comment]
Huh. The last I heard, they have their own copy. Maybe I misunderstood. But like I said, as long as you don't need to convert between mag and linear flux, Astropy should be okay for now.

**Comment by nmearl** [comment]
>  understand that the JWST ETC does not use pysynphot at all.

Pysynphot is used in the ETC. The background and SED classed in particular makes heavy use of it. Also, a lot of rebinning happens, and pysynphot is used there as well. I'm currently debugging a front-end error propagated up by a pysynphot error in the engine.

This is assuming you mean the new JWST ETC a.k.a. "Pandeia".

**Comment by ibusko** [comment]
Right, I had a generic “no pysynphot in Pandeia” in my recollection, but in fact Pandeia does not use pysynphot *for throughput calculations*. That is, anything that requires the graph table. So the spectrum-related functionality is being used after all. 

Thanks for the clarification!

**Comment by pllim** [comment]
A bit off-topic but... You don't really need graph tables to generate throughputs in pysynphot. You can read in the components yourself and multiply them to create final throughput as well. :)

**Comment by nmearl** [comment]
Implemented in 37956fef9abb3d76d60ccc90217bcc687fa992e0.
